File: Freedom/微信/发现/WXGameViewController.m
//  TLGameViewController.m
//  Freedom
//  Created by Super on 16/3/4.
#import "WXGameViewController.h"
@implementation WXGameViewController
- (void)viewDidLoad{
    [super viewDidLoad];
    [self setUseMPageTitleAsNavTitle:NO];
    [self setShowLoadingProgress:NO];
    [self setDisableBackButton:YES];
    
    [self.navigationItem setTitle:@"游戏"];
    [self setUrl:@"http://m.le890.com"];
    
    UIBarButtonItem *rightBarButton = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"nav_setting"] style:UIBarButtonItemStylePlain target:self action:@selector(rightBarButtonDown:)];
    [self.navigationItem setRightBarButtonItem:rightBarButton];
    
    [SVProgressHUD showWithStatus:@"加载中"];
}
- (void)viewWillDisappear:(BOOL)animated{
    [super viewWillDisappear:animated];
    if ([SVProgressHUD isVisible]) {
        [SVProgressHUD dismiss];
    }
}
#pragma mark - Delegate -
- (void)webView:(WKWebView *)webView didFinishNavigation:(WKNavigation *)navigation{
    if ([SVProgressHUD isVisible]) {
        [SVProgressHUD dismiss];
    }
    [super webView:webView didFinishNavigation:navigation];
}
#pragma mark - Event Response
- (void) rightBarButtonDown:(UIBarButtonItem *)sender{
    
}
@end

File: Freedom/微信/发现/WXGameViewController.swift
//
//  WXGameViewController.swift
//  Freedom

import Foundation
import WebKit
//import XCarryOn
class WXGameViewController: WXWebViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        useMPageTitleAsNavTitle = false
        showLoadingProgress = false
        disableBackButton = true
        self.url = "http://m.le890.com"
        navigationItem.title = "游戏"
        let rightBarButton = UIBarButtonItem(image: Image.setting.image, style: .plain, target: self, action: #selector(self.rightBarButtonDown(_:)))
        navigationItem.rightBarButtonItem = rightBarButton
        XHud.show()
    }
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        XHud.hide()
    }
    override func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
        XHud.hide()
        super.webView(webView, didFinish: navigation)
    }
    @objc func rightBarButtonDown(_ sender: UIBarButtonItem) {
    }
}
